src/coreclr/jit/fgprofile.cpp

```diff
@@ -4643,10 +4651,20 @@ bool Compiler::fgDebugCheckProfileWeights(ProfileChecks checks)
     //
     for (BasicBlock* const block : Blocks())
     {
-        if (!block->hasProfileWeight() && !checkAllBlocks)
+        if (!block->hasProfileWeight())
         {
-            unprofiledBlocks++;
-            continue;
+            if (checkProfileFlag && (block->bbPreds != nullptr))
```

Comment:
Can you explain the exclusion for blocks with no preds?

Reply:
For a few phases after `fgAddInternal`, merged return blocks are unreachable, and thus haven't had the `BBF_PROF_WEIGHT` flag propagated into them. I suppose I could've explicitly checked for that case here, or disabled the checks for those phases, though this should cover other cases where blocks have been created, but they don't have flow into them yet.

I guess we're losing some coverage by not catching method/region entry blocks that aren't flagged, though it seems very unlikely that these blocks would lack profile weights while their successors do have them(?)

Reply:
Ah, ok... annoying but understandable that there are going to be some exclusions. We might be able to narrow the exclusions somewhat but I wouldn't bother.